test_files/unit_tests/test_symmetric_site.py

```
@@ -0,0 +1,242 @@
+
+@pytest.mark.parametrize("symmetry", [2.0, "2", None, True, False])
+def test_init_rejects_non_integer_symmetry(base_site, symmetry):
+    """Verify symmetry must be an integer because it specifies a discrete number of sites."""
+    with pytest.raises(TypeError, match="symmetry must be an integer"):
+        SymmetricSite(symmetry, base_site)
+
+
+@pytest.mark.parametrize("symmetry", [1, 0, -1])
+def test_init_rejects_symmetry_below_two(base_site, symmetry):
+    """Verify symmetry below two fails because a onefold site should use Site directly."""
+    with pytest.raises(ValueError, match="symmetry must be greater than one."):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)